lib/galaxy/datatypes/binary.py

```diff
@@ -810,6 +812,83 @@ def display_peek(self, dataset):
 
 
+class Biom2(H5):
+    """
+    Class describing a biom2 file (http://biom-format.org/documentation/biom_format.html)
+    """
+    MetadataElement(name="id", default=None, desc="table id", readonly=True, visible=True, no_value=None)
+    MetadataElement(name="format_url", default=None, desc="format-url", readonly=True, visible=True, no_value=None)
+    MetadataElement(name="format_version", default=None, desc="format-version", readonly=True, visible=True, no_value=None)
+    MetadataElement(name="format", default=None, desc="format", readonly=True, visible=True, no_value=None)
+    MetadataElement(name="type", default=None, desc="table type", readonly=True, visible=True, no_value=None)
+    MetadataElement(name="generated_by", default=None, desc="generated by", readonly=True, visible=True, no_value=None)
+    MetadataElement(name="creation_date", default=None, desc="creation date", readonly=True, visible=True, no_value=None)
+    MetadataElement(name="nnz", default=None, desc="nnz: The number of non-zero elements in the table", readonly=True, visible=True, no_value=None)
+    MetadataElement(name="shape", default=None, desc="shape: The number of rows and columns in the dataset", readonly=True, visible=True, no_value=None)
+
+    file_ext = "biom2"
+    edam_format = "format_3746"
+
+    def sniff(self, filename):
+        """
+        >>> from galaxy.datatypes.sniff import get_test_fname
+        >>> fname = get_test_fname( 'biom2_sparse_otu_table_hdf5.biom' )
+        >>> Biom2().sniff( fname )
+        True
+        >>> fname = get_test_fname( 'test.mz5' )
+        >>> Biom2().sniff( fname )
+        False
+        >>> fname = get_test_fname( 'wiggle.wig' )
+        >>> Biom2().sniff( fname )
+        False
+        """
+        if super(Biom2, self).sniff(filename):
+            try:
+                f = h5py.File(filename)
+                attributes = list(dict(f.attrs.items()))
+                required_fields = ['id', 'format-url', 'type', 'generated-by', 'creation-date', 'nnz', 'shape']
+                return set(required_fields).issubset(attributes)
+            except Exception:
+                return False
+        return False
+
+    def set_meta(self, dataset, overwrite=True, **kwd):
+        super(Biom2, self).set_meta(dataset, overwrite=overwrite, **kwd)
+        try:
+            f = h5py.File(dataset.file_name)
+            attributes = dict(f.attrs.items())
+
+            dataset.metadata.id = attributes['id']
+            dataset.metadata.format_url = attributes['format-url']
+            if 'format-version' in attributes:  # biom 2.1
+                dataset.metadata.format_version = '.'.join(map(str, list(attributes['format-version'])))
+            elif 'format' in attributes:  # biom 2.0
+                dataset.metadata.format = attributes['format']
+            dataset.metadata.type = attributes['type']
+            dataset.metadata.shape = str(list(attributes['shape']))
```

Comment:
is this supposed to be a tuple? Like it is here the brackets will end up in the metadata as well, isn't it?

Reply:
indeed, but my concern was that anything that's not a string was not being displayed in the GUI when looking at the metadata, and users might quite like to see that information. But sounds like that should proably be fixed elsewhere then (same for nnz, it is an int but doesn't show up in interface unless I cast it to string)